controllers/booksController.js

```diff
@@ -0,0 +1,137 @@
+const { Op } = require("sequelize");
+const fs = require("fs");
+const ValidationChecker = require("./ValidationChecker");
+
+class BooksController extends ValidationChecker {
+  constructor(db) {
+    super();
+    this.bookModel = db.book;
+    this.photoModel = db.photo;
+    this.categoryModel = db.category;
+    this.donationModel = db.donation;
+    this.userModel = db.user;
+    this.sequelize = db.sequelize;
+  }
+
+  async insertBook(req, res) {
+    const { categories, email, ...data } = JSON.parse(req.body.data);
+    const t = await this.sequelize.transaction();
+    try {
+      this.checkStringFromBody(email, "email");
+      this.checkArray(categories, "categories");
+
+      const book = await this.bookModel.create(data, { transaction: t });
+      for (const [index, { path }] of req.files.entries()) {
+        const photoBinaryData = fs.readFileSync(path);
+        await this.photoModel.create(
+          {
+            index: index,
+            bookId: book.id,
+            file: photoBinaryData,
+          },
+          { transaction: t }
+        );
+        fs.unlinkSync(path);
+      }
+
+      const bookCategory = await this.categoryModel.findAll({
```

Comment:
```suggestion
      const bookCategories = await this.categoryModel.findAll({
```

findAll returns a list, so we use plural